Tests/ApolloCodegenTests/CodeGeneration/Templates/SelectionSet/SelectionSetTemplateTests.swift

```diff
@@ -1487,787 +1487,753 @@ class SelectionSetTemplateTests: XCTestCase {
-        .field("__typename", String.self),
-        .include(if: "a", [
-          .field("fieldA", String.self),
-          .field("fieldB", String.self),
-          .inlineFragment(AsPetIfA.self),
-          .inlineFragment(IfA.self),
-        ]),
+        .field("species", String?.self),
       ] }
     """
 
     // when
     try buildSubjectAndOperation()
-    let allAnimals = try XCTUnwrap(
-      operation[field: "query"]?[field: "allAnimals"] as? IR.EntityField
+    let allAnimals_asDog = try XCTUnwrap(
+      operation[field: "query"]?[field: "allAnimals"]?[as: "Dog"]
     )
 
-    let actual = subject.render(field: allAnimals)
+    let actual = subject.render(inlineFragment: allAnimals_asDog)
 
     // then
-    expect(actual).to(equalLineByLine(expected, atLine: 7, ignoringExtraLines: true))
+    expect(actual).to(equalLineByLine(expected, atLine: 8, ignoringExtraLines: true))
   }
 
-  func test__render_selections__givenFragmentWithNonMatchingTypeAndInclusionCondition_rendersTypeCaseSelectionWithInclusionCondition() throws {
+  func test__render_selections__givenNestedDeferredInlineFragments_rendersNestedDeferredFragmentSelections() throws {
     // given
     schemaSDL = """
     type Query {
       allAnimals: [Animal!]
     }
 
-    type Animal {
-      string: String!
-      int: Int!
+    interface Animal {
+      id: String
+      species: String
+      genus: String
     }
 
-    type Pet {
-      string: String!
-      int: Int!
+    type Dog implements Animal {
+      id: String
+      species: String
+      genus: String
+      name: String
+      friend: Animal
+    }
+
+    type Cat implements Animal {
+      id: String
+      species: String
+      genus: String
     }
     """
 
     document = """
-    query TestOperation($a: Boolean!) {
+    query TestOperation {
       allAnimals {
-        ...FragmentA @include(if: $a)
+        __typename
+        id
+        ... on Dog @defer(label: "one") {
+          friend {
+            ... on Cat @defer(label: "two") {
+              species
+            }
+          }
+        }
       }
     }
-
-    fragment FragmentA on Pet {
-      int
-    }
     """
 
-    let expected = """
+    let expectedOne = """
       public static var __selections: [ApolloAPI.Selection] { [
-        .field("__typename", String.self),
-        .include(if: "a", .inlineFragment(AsPetIfA.self)),
+        .deferred(One.self, label: "one"),
+      ] }
+    """
+    let expectedTwo = """
+      public static var __selections: [ApolloAPI.Selection] { [
+        .deferred(Two.self, label: "two"),
       ] }
     """
 
     // when
     try buildSubjectAndOperation()
-    let allAnimals = try XCTUnwrap(
-      operation[field: "query"]?[field: "allAnimals"] as? IR.EntityField
+    let allAnimals_asDog = try XCTUnwrap(
+      operation[field: "query"]?[field: "allAnimals"]?[as: "Dog"]
+    )
+    let allAnimals_asDog_asCat = try XCTUnwrap(
+      allAnimals_asDog[as: "One"]?[field: "Friend"]?[as: "Cat"]
```

Comment:
1. The `as` subscript is not going to work here, because it will be looking for an actual type named "One" to match against the runtime type of the object. We'll need a new subscript for deferred fragments.

2. friend should be lowercase
```suggestion
      allAnimals_asDog[as: "One"]?[field: "friend"]?[as: "Cat"]
```

Reply:
> The as subscript is not going to work here, because it will be looking for an actual type named "One" to match against the runtime type of the object. We'll need a new subscript for deferred fragments.

Yes - this will get done as part of making the tests pass.